src/main.js

```diff
@@ -1,8 +1,38 @@
 import { getPokemonList } from './api/index.js';
 import { createDropdownTypes, createPokemonList } from './dom/index.js';
 
-const addListeners = () => {
-	// Code goes here
+const filterCards = () => {
+	// Define variable to select the Dropdown from the DOM.
+	const dropdownType = document.getElementById("dropdown-types"); 
+
+	// Add the event listener to the Dropdown to catch when the selection changes and filter the list based on the selection
+	dropdownType.addEventListener('change', (event) => {
+		// Constant with the Card list item DOM element
+		const cardsList = document.querySelectorAll(".card-list__item");
+		// Variable to store the value selected in the dropdown
+		let currentSelection = dropdownType.value;
+		// Variable to store the type of the pokemon that is being evaluated
+		let currentPokemonType = "";
+		
+		// Logic to determine wheter the element should have the hide class or not
+		// If the selection is "all", the hide class should be removed from all the elements
+		// If the selection is a type, the "hide"class should be added to the pokemons that do not have that type
+		// and should be removed from the pokemons that have that type.
+		if (currentSelection == "all") {
+			cardsList.forEach(element => {
+					element.classList.remove("hide");
+			});
+		} else {
+			cardsList.forEach(element => {
+				currentPokemonType = element.attributes[0].value;
```

Comment:
Try using the function `getAttribute` of the element to get the value of `data-types`. It's safer than accessing a fixed position of the array. If in the future the position of the attribute changes, you would need to update the index here.